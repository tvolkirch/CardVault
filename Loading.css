#spinner-background
{
    width: 1px;
    height: 1px;
    position: fixed;
    background-color: white;
    z-index: 10;
}

#spinner
{
    animation: spin 1s infinite;
    animation-timing-function: linear;
}
    
@keyframes spin
{
    0% { transform: rotate(0); }
    100% { transform: rotate(360deg); }
}
