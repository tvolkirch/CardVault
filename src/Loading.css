#spinner-background
{
    width: 50px;
    height: 0.25rem;
    position: fixed;
    display: flex;
    justify-content: center;
    background-color: red;
    border: 5px solid #111;
    border-radius: 50%;
    z-index: 10;
    animation: spin 0.3s infinite;
    animation-timing-function: linear;
}

#inner-spinner
{
    position: relative;
    top: -25px;
    height: 50px;
    width: 0.25rem;
    background-color: red;
    border: 5px solid #111;
    border-radius: 50%;
}

@keyframes spin
{
    0% { transform: rotate(0); }
    100% { transform: rotate(360deg); }
}
