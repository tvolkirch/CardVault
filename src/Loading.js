import React from "react";

import "./Loading.css";

function Loading(props)
{
  var ctop = props.top || "";
  var cleft = props.left || "";
  var cbottom = props.bottom || "";
  var cright = props.right || "";

  if ( ctop !== "" )
  {
    ctop = ctop + "px";
  }

  if ( cleft !== "" )
  {
    cleft = cleft + "px";
  }

  if ( cbottom !== "" )
  {
    cbottom = cbottom + "px";
  }

  if ( cright !== "" )
  {
    cright = cright + "px";
  }
  
  const spinPosition = { top: ctop, left: cleft, bottom: cbottom, right: cright };

   return (
<div id="spinner-background" style={spinPosition}>&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;
<div id="inner-spinner">&nbsp;</div>&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;
</div>
   );
}

export default Loading;