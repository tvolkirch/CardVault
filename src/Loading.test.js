import React from "react";
import renderer from "react-test-renderer";

import Loading from "./Loading";

it("Consistently renders the spinner", () => {

   const spinner = renderer
      .create(<Loading top={25} left={100} bottom={100} right={25} />)
      .toJSON();

   expect(spinner).toMatchSnapshot();
});
