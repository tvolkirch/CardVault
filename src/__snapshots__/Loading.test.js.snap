// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Consistently renders the spinner 1`] = `
<div
  id="spinner-background"
  style={
    Object {
      "bottom": "100px",
      "left": "",
      "right": "25px",
      "top": "",
    }
  }
>
        
  <div
    id="inner-spinner"
  >
     
  </div>
        
</div>
`;
